src/vectorchord/usage/indexing.md

```diff
@@ -39,29 +41,85 @@ The table below shows the operator classes for types and operator in the index.
+## Important Usage Recommendations
+
+When dealing with large datasets (>1M vectors), please follow these guidelines for optimal performance:
+
+1. First insert all vectors into the table before building the index
+2. Select an appropriate number of lists (`build.internal.lists` parameter) based on your dataset size
+3. Failure to follow these steps may result in significantly increased query latency
+
+## Indexing Options
+
+#### `residual_quantization`
+    
+- Description: This index parameter determines whether residual quantization is used. If you not familiar with residual quantization, you can read this [blog](https://drscotthawley.github.io/blog/posts/2023-06-12-RVQ.html) for more information. Shortly, residual quantization is a technique that improves the accuracy of vector search by quantizing the residuals of the vectors.
+- Type: boolean
+- Default: `false`
+- Example:
+    - `residual_quantization = false` means that residual quantization is not used.
+    - `residual_quantization = true` means that residual quantization is used.
+- Note: set `residual_quantization` to `true` if your model generates embeddings where the metric is Euclidean distance. This option only works for L2 distance. Using it with other distance metrics will result in an error in building.
+
+### Internal Build Parameters
+
+All build parameters are specified in the `options` using [TOML: Tom's Obvious Minimal Language](https://toml.io/) string of the index. The following parameters are available:
+
+#### `build.internal.lists`
+    
+- Description: This index parameter determines the hierarchical structure of the vector space partitioning.
+- Type: list of integers
+- Default: `[]`
+- Example:
+    - `build.internal.lists = []` means that the vector space is not partitioned.
+    - `build.internal.lists = [4096]` means the vector space is divided into $4096$ cells.
+    - `build.internal.lists = [4096, 262144]` means the vector space is divided into $4096$ cells, and those cells are further divided into $262144$ smaller cells.
+- Note: The index partitions the vector space into multiple Voronoi cells using centroids, iteratively creating a hierarchical space partition tree. Each leaf node in this tree represents a region with an associated list storing vectors in that region. During insertion, vectors are placed in lists corresponding to their appropriate leaf nodes. For queries, the index optimizes search by excluding lists whose leaf nodes are distant from the query vector, effectively pruning the search space. The `lists` option should be no less than `4 * sqrt(vectors)`, `vectors` is the number of vectors in the table. 
```

Comment:
```suggestion
- Note: The index partitions the vector space into multiple Voronoi cells using centroids, iteratively creating a hierarchical space partition tree. Each leaf node in this tree represents a region with an associated list storing vectors in that region. During insertion, vectors are placed in lists corresponding to their appropriate leaf nodes. For queries, the index optimizes search by excluding lists whose leaf nodes are distant from the query vector, effectively pruning the search space. The `lists` option should be no less than $4 * \sqrt{N}$, where $N$ is the number of vectors in the table. 
```

If a list of length 1 is recommended here, please specify it first. Do not say that it's a list of numbers while saying that a number should be used as its value.

Reply:
ok